app/Http/Controllers/BookController.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Book;
+use App\Models\Category;
+use App\Models\Author;
+use App\Models\Publisher;
+use App\Http\Requests\BookRequest;
+use App\Components\Recusive;
+
+class BookController extends Controller
+{
+    private $book;
+    private $category;
+    private $author;
+    private $publisher;
+
+    public function __construct(
+        Book $book, 
+        Category $category, 
+        Author $author, 
+        Publisher $publisher
+        ) {
+        $this->book = $book;
```

Comment:
```suggestion
        Publisher $publisher
    ) {
        $this->book = $book;
```
sửa nốt 1 tý này đi em, lần sau ko sai nữa